### PR TITLE
Add default timeouts for Analytics API and LMS

### PR DESCRIPTION
**Pull request by dan-f**
I added django settings to dictate default timeouts when waiting on both the LMS and Analytics API, respectively. These timeouts can be overridden both in code as well as in configuration.

@fredsmith this will require coordination with DevOps as I mentioned last time we chatted. Reasonable timeouts should be set in configuration.

@dsjen please review, assuming tests pass this time...

[AN-6250](https://openedx.atlassian.net/browse/AN-6250)

### CODE REVIEW

**Comment by dan-f** [comment]
Looking at the failure now...

File: AUTHORS [hunk 1]
```diff
@@ -2,3 +2,4 @@ Clinton Blackburn <cblackburn@edx.org>
 Dennis Jen <djen@edx.org>
 Carlos Andrés Rocha <rocha@edx.org>
 Braden MacDonald <braden@opencraft.com>
+Daniel Friedman <dfriedman@edx.org>
```
Comment (dsjen) [inline AUTHORS:5]: Finally!

File: analytics_dashboard/courses/views/__init__.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 from django.views.generic import TemplateView
 from opaque_keys.edx.keys import CourseKey
 import requests
-from slumber.exceptions import (HttpClientError, SlumberBaseException)
+from edx_rest_api_client.exceptions import (HttpClientError, SlumberBaseException)
```
Comment (dsjen) [inline analytics_dashboard/courses/views/__init__.py]: Moving this with the edx imports in alphabetical order would follow the pattern a bit better (not that we've been consistent...).

File: requirements/base.txt [hunk 3]
```diff
@@ -14,7 +14,7 @@ django-waffle==0.10			# BSD
 libsass==0.5.1              # MIT
 
 logutils==0.3.3				# BSD
-slumber==0.6.2
+edx-rest-api-client==1.2.1              # Apache
```
Comment (dsjen) [inline requirements/base.txt]: Putting this in order would be helpful.

File: analytics_dashboard/courses/tests/test_views/__init__.py [hunk 4]
```diff
@@ -28,11 +28,14 @@ class CourseAPIMixin(SwitchMixin):
...
     Mixin with methods to help mock the course API.
     """
-
     COURSE_API_COURSE_LIST = {'next': None, 'results': [
         {'id': course_key, 'name': 'Test ' + course_key} for course_key in [DEMO_COURSE_ID, DEPRECATED_DEMO_COURSE_ID]
     ]}
 
+    def setUp(self, *args, **kwargs):
+        super(CourseAPIMixin, self).setUp(*args, **kwargs)
+        self.toggle_switch('enable_course_api', True)
```
Comment (dsjen) [inline analytics_dashboard/courses/tests/test_views/__init__.py:37]: Nice!

**Comment by dsjen** [comment]
There are more deleted lines than added--fun!  Thanks for chipping away at our tech debt and this looks good to me! :+1:

Let me know if you need any help with the failures.

**Comment by dan-f** [comment]
@dsjen tests finally passed. Merging now.

_edit_ j/k not merging. Need to touch base with devops regarding configuration.

**Comment by dsjen** [comment]
:dancer: :dancers: :clap: :tophat:

**Pull request merged**